### PR TITLE
f32/f64/f16: zero DotProductBatch on empty vec, add scalar IRFFT pack below nfft 32

### LINKED ISSUE #295: f32/f64 inverse STFT: follow-ups from the IRFFT/ISTFT vectorization
Follow-ups deferred from #293 (vectorized IRFFT pack, MulAdd overlap-add and block normalization). None blocks correctness; each needs its own measurement.

**1. IRFFT is slower below nfft 32.** The vector pack replaced one scalar loop with a split loop, two `clear` calls, `RealFFTUnpack`, two `Scale` calls and `Interleave2`, none of which inline, and below the SIMD dispatch thresholds they run the Go fallbacks. Measured against the old scalar path: nfft 4 is about +65%, nfft 8 about +35%, nfft 16 up to +18% (a fixed 15ns to 35ns per call on an i7-1260P and a Raspberry Pi 5). nfft 32 and above is faster, and 512 to 2048 is 20% to 49% faster. Option: keep a scalar pack and output loop when `half` is below a small per-plan cutoff, measured per tier with `BenchmarkIRFFT` (extend it down to nfft 4 and 8). Any cutoff must keep a short spectrum bit-identical to its zero-filled form within a plan (`TestIRFFTShortInputs`).

**2. Hoist per-frame constant work out of ISTFT.** Each frame runs IRFFT's `Scale(re, 1/half)` and `Scale(im, -1/half)` and then `MulAdd` by the window, although the scale and the alternating sign are constant per call. An internal unscaled IRFFT plus a per-call pre-scaled, sign-alternated window (one extra nfft scratch buffer) would cut the per-frame passes from four to two. This changes rounding, which the documented tolerance-stable contract allows.

**3. Fewer Div calls in the block normalization.** `normalizeISTFT` calls `Div` once per hop-sized block; at small hops that is one dispatch per few samples. Tiling the periodic table across `p.frame[:(nfft/hop)*hop]` and stepping by that length stays bit-identical and cuts the call count by `nfft/hop`. Measure with an ISTFT hop sweep (hop 1, 4, 16, nfft/4).

**4. Docs and test hygiene.** `doc.go` lists only `NewSTFTPlan, STFT, STFTPower, STFTPowerInto, NumFrames` for STFTPlan and omits `RFFT`, `IRFFT`, `ISTFT`, `NumBins` and `NFFT`. The f32 package doc names `FMA` and `AddScaled` as the fused operations but not `MulAdd`. `TestISTFTAllocFree` discards the ISTFT return value, so an early-return regression would pass as zero allocations; assert the count inside the closure.

### LINKED ISSUE #303: f32/f64 DotProductBatch leaves results stale on an empty vec (i8 zeroes them)
Found during the review of #302 (i8.DotProductBatch).

f32.DotProductBatch (f32/f32.go, dotProductBatch32) and f64.DotProductBatch return early on `n == 0 || len(vec) == 0`, leaving `results[:n]` untouched. When vec is empty and the caller reuses a results buffer, it gets back stale values from a previous call rather than the dot products of an empty vector.

The mathematically correct result for an empty vec is 0 for every row (the dot product of an empty vector is 0), which is also what an empty row already yields. i8.DotProductBatch (added in #302) zeroes `results[:n]` on an empty vec for exactly this reason, so the three sibling packages now disagree: i8 zeroes, f32/f64 leave stale memory.

Proposed fix: on an empty vec, zero `results[:n]` in f32 and f64 before returning, matching i8 and the empty-row identity. Alternatively, document the divergence explicitly in all three, but zeroing is the less surprising behavior. Add an f32/f64 test that pre-fills results with a sentinel and asserts they become 0 for an empty vec, mirroring TestDotProductBatchEmptyVec.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Two follow-ups deferred from #302 and #293, both in the float DSP packages.

**#303: `DotProductBatch` no longer leaves stale results on an empty vec.** The public `f32`/`f64` `DotProductBatch` returned early on an empty `vec`, leaving `results[:n]` holding whatever the caller's buffer contained from a previous call. The dot product of an empty vector is 0, which is already what an empty row yields and what `i8.DotProductBatch` writes. Dropping the `len(vec)==0` short-circuit lets the kernel run and zero `results[:n]`, since every internal kernel (f32/f64/f16 across amd64/arm64/go) already zeroes each row on `vecLen==0`. The same one-line change is applied to `f16` too, which was the remaining sibling still on the old behavior, so all four numeric packages (`i8`, `f16`, `f32`, `f64`) now share the contract. Each package gains a `TestDotProductBatchEmptyVec` that pins zeroing of `results[:n]` (for both `nil` and empty `vec`) and that `results[n:]` stays untouched.

This is an observable public-API behavior change on the empty-vec edge: a caller reusing a `results` buffer now gets zeros back instead of stale values. Worth a line in the release notes.

**#295 item 1: a fused scalar IRFFT pack for small transforms.** Below nfft 32 the vectorized inverse pack lost to the old scalar path, because `packInverse` (shared by `IRFFT` and ISTFT's `irfftFrameUnscaled`) calls the non-inlined `RealFFTUnpack`, `Scale` and `Interleave2` Go fallbacks plus two scratch clears and a split load, none of which pay off at tiny sizes. A new `packInverseScalar` computes the same `conj(C)` in one fused loop, and `IRFFT`/`irfftFrameUnscaled` write the small frame with scalar output/interleave loops, gated by a `half < 16` cutoff (nfft < 32). The scalar pack inlines the `realFFTUnpack` Go-fallback even/odd math bin for bin and mirrors `packInverse`'s deliberate `0-imag` signed-zero convention, so it is bit-identical to the fallback wherever the vector path already used it (half <= 8 on amd64 f32, half <= 4 on amd64 f64) and preserves the short-spectrum == zero-filled contract; elsewhere it stays within the documented tolerance-stable IRFFT/ISTFT contract. Since `half` is always a power of two, nfft >= 32 keeps the vector path untouched.

`TestIRFFTShortInputs` now sweeps nfft {8, 16, 32}, so the scalar pack's short-spectrum guards and signed-zero handling are exercised bit-exactly, not just the vector path at nfft 32.

`BenchmarkIRFFT`, count=10 (arm64 core-pinned on a Cortex-A76):

| size | f32 amd64 | f64 amd64 | f32 arm64 | f64 arm64 |
|------|-----------|-----------|-----------|-----------|
| nfft 4 | -45% | -44% | -37% | -39% |
| nfft 8 | -34% | -32% | -25% | -28% |
| nfft 16 | -15% | -11% | -3% | -16% |
| nfft >= 32 | unchanged | unchanged | unchanged | unchanged |

Zero allocations are preserved on all paths.

## Related Issues

Fixes #303
Fixes #295

## Test Plan

- [x] `go test ./...` green on amd64 (13 packages)
- [x] Full `f16`/`f32`/`f64` suites green natively on arm64 (Cortex-A76)
- [x] New `TestDotProductBatchEmptyVec` (f32/f64/f16) and expanded `TestIRFFTShortInputs` (nfft 8/16/32) pass, including the bit-exact short == zero-filled check on the scalar path
- [x] `golangci-lint run` clean; `AllocsPerRun == 0` assertions hold
- [x] `BenchmarkIRFFT` before/after measured on amd64 and arm64

## Summary by CodeRabbit

* **Bug Fixes**
  * Empty-vector batch dot products now correctly write zero results for all processed rows while preserving values beyond the processed range.
  * Inverse FFT processing now correctly handles small transforms, truncated or empty spectra, short destinations, and DC/Nyquist values.

* **Performance**
  * Small inverse FFT transforms use optimized scalar processing while larger transforms retain existing vectorized processing.

* **Tests**
  * Expanded coverage for empty vectors, short spectra, destination limits, nil destinations, zero-extension, and small FFT sizes.

### CODE REVIEW

File: f32/stft_test.go [hunk 1]
```diff
@@ -1292,58 +1292,66 @@ func TestRFFTIRFFTRoundTripF32(t *testing.T) {
...
+			}
+			p.IRFFT(got, nil)
+			for i, v := range got {
+				if v != 0 {
+					t.Fatalf("sample %d: empty spec gave %g, want 0", i, v)
+				}
+			}
+		})
 	}
 }
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/stft_test.go:1372]: Done in 35b2050. Added nfft=4 to both sweeps (the minimal scalar transform, half=2, where the single interior bin is its own mirror). For the empty-spectrum check I went slightly further than a plain +0 bit compare: IRFFT of an all-zero spectrum legitimately writes -0.0 into the odd output samples (dst[2j+1] = -im*scale with im == +0), so asserting bits == +0 would false-fail. It now asserts IRFFT(nil) is bit-identical to IRFFT of an explicit zero-filled spectrum (signed zeros aligned) and zero in magnitude, which still catches stale values and any signed-zero divergence.
